src/main/java/org/apache/commons/csv/CSVRecord.java

```diff
@@ -272,5 +337,52 @@ public String toString() {
     String[] values() {
         return values;
     }
+    
+    /**
+     * Return a CSVRecord with the given column value set.
+     * 
+     * @param name
+     *            the name of the column to set.
+     * @param value
+     * 			  The new value to set
+     * @throws IllegalStateException
+     *             if no header mapping was provided
+     * @throws IllegalArgumentException
+     *             if {@code name} is not mapped or if the record is inconsistent
+     * @return A mutated CSVRecord
+     */
+    public CSVRecord withValue(String name, String value) {    	
+    	CSVRecord r = mutable();
+    	r.put(name,  value);
+    	return r;
```

Comment:
Note that here the javadoc leaves mutability of the returned CSVRecord undefined, this implementation will in fact always return a mutable CSVRecord (as can be verified with `isMutable()`)

If someone did `record.withValue(1, "b").immutable()` it would cause two array copies. But `record.withValue(1, "b").withValue(2, "c") would just make one array copy (or 0 if withMutableRecords was set in format).